sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -55,7 +56,7 @@ object SQLConf {
   val WAREHOUSE_PATH = SQLConfigBuilder("spark.sql.warehouse.dir")
     .doc("The default location for managed databases and tables.")
     .stringConf
-    .createWithDefault("${system:user.dir}/spark-warehouse")
+    .createWithDefault("${system:user.dir}" + File.separator + "spark-warehouse")
```

Comment:
`Path` normalizes `backslashes` to forward slashes on Windows. Maybe it is safe to use `File.separator`. If anybody hit an issue in Windows, we can make a change then. : )